coffea/nanoevents/factory.py

```diff
@@ -69,14 +69,18 @@ def from_file(
             metadata : dict, optional
                 Arbitrary metadata to add to the `base.NanoEvents` object
             uproot_options : dict, optional
-                Any options to pass to ``uproot4.open``
+                Any options to pass to ``uproot.open``
         """
         if not issubclass(schemaclass, BaseSchema):
             raise RuntimeError("Invalid schema type")
         if isinstance(file, str):
-            tree = uproot4.open(file, **uproot_options)[treepath]
-        elif isinstance(file, uproot4.reading.ReadOnlyDirectory):
+            tree = uproot.open(file, **uproot_options)[treepath]
+        elif isinstance(file, uproot.reading.ReadOnlyDirectory):
             tree = file[treepath]
+        elif "<class 'uproot.rootio.ROOTDirectory'>" == str(type(file)):
+            raise RuntimeError("The file intance (%s) is an uproot3 type, but this module is only compatible with uproot4 or higher" % (str(type(file))))
```

Comment:
`"The file instance (%r) is an uproot3 type, but this module is only compatible with uproot4 or higher" % file)`